Split monolith.

Novacancy531 opened the pull request:
Разбил приложение на 2 сервиса.
Добавил функциональность request.
Добавил тесты.

Novacancy531 committed:
Реализован Request функционал.

Novacancy531 committed:
Разделение монолита.

Novacancy531 committed:
Файлы сервисов

Novacancy531 committed:
Тесты и правки.

Novacancy531 committed:
revert checkstyle.xml

Novacancy531 committed:
Фикс названия ItemControllerTest.

Novacancy531 committed:
Фикс названия ItemControllerTest.

Novacancy531 committed:
Merge remote-tracking branch 'origin/add-item-requests-and-gateway' into add-item-requests-and-gateway

Novacancy531 committed:
Фикс ItemControllerTest.

Novacancy531 committed:
Больше тестов.

Novacancy531 committed:
Еще больше тестов.

Novacancy531 committed:
Замена бд на h2

Novacancy531 committed:
Заменил на файл из учебного репозитория.

Novacancy531 committed:
Включил автосоздание таблиц

VadimZharkov commented on gateway/src/main/java/ru/practicum/shareit/booking/BookingClient.java:
```diff
@@ -0,0 +1,61 @@
+
+    private static final String API_PREFIX = "/bookings";
+
+    @Autowired
```
`@Autowired` перед конструктором можно не писать.

VadimZharkov commented on server/src/main/java/ru/practicum/shareit/request/service/RequestServiceImpl.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    @Override
+    public List<RequestWithItems> getRequests(Long userId) {
```
По ТЗ запросы сортируются по дате создания от более новых к более старым.

VadimZharkov commented on server/src/test/java/ru/practicum/shareit/booking/model/BookingTest.java:
```diff
@@ -0,0 +1,47 @@
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BookingTest {
```
Не понятно для чего модели тестируются. По ТЗ нужно реализовать тесты для Dto с помощью `@JsonTest`

VadimZharkov reviewed (changes requested):
Хорошая работа, нужно исправить недочеты.

Novacancy531 committed:
Исправлены замечания.
Добавлены @JsonTest.
Добавлена сортировка для запросов вещей.
Убран излишний autowired.

Novacancy531 committed:
Фикс checkstyle
